### PR TITLE
EQL: Replace [`] with ['] for identifiers

### PR DESCRIPTION
**Pull request by matriv**
Since `'` is no longer supported for defining string literals,
we replace the backquote [`] with single quote ['] for the definition of
identifiers. The backquote is not a commonly used symbol in languages
and many times, especially with some fonts, can easily be mistaken and
cause confusion.

Closes: #62539

### CODE REVIEW

File: x-pack/plugin/eql/src/main/antlr/EqlBase.g4 [hunk 1]
```diff
@@ -191,14 +191,12 @@ PIPE: '|';
 
 
 ESCAPED_IDENTIFIER
-    : '`' (~'`')* '`'
+    : '\'' (~'`')* '\''
```
Comment (rw-access) [inline x-pack/plugin/eql/src/main/antlr/EqlBase.g4:194]: We haven't agreed on this.

https://github.com/elastic/elasticsearch/pull/62539#pullrequestreview-491521943

**Review by rw-access — CHANGES_REQUESTED** [review]
Please don't merge until we reach an agreement. Related note:in this PR as well: https://github.com/elastic/elasticsearch/pull/62539#pullrequestreview-491521943.

Also EQL _in Elasticsearch_ is experimental, but EQL for Endgame is not. For EQL in Endgame users migrating to Elasticsearch, this is would make this not only a breaking change, but a quiet one, since existing syntax is assigned new purpose.

**Comment by matriv** [comment]
@rw-access I agree with you, since there is no way (grammar-wise) to keep both the error message for the single quotes around strings and their new usage around identifiers. Personally, I'd keep the next release with the prohibition of single quotes and introduce the new usage in the release after that.

**Comment by matriv** [comment]
Let me add some more thoughts.

- The single quote `'` is currently used rarely to identify strings
- If a user issues a query with `'` around a string literal, e.g.: `process where a = 'foo'` then the foo will be parsed as a identifier and:
  - if `foo` is a valid identifier, an exception will be thrown since we cannot have comparisons with identifiers both to the left and to the right.
  - if `foo` is an invalid identifier, an exception will be thrown stating this.

This doesn't seem as a dangerous behaviour since it doesn't lead to a correct query with wrong behaviour.

On the other hand, for me it seems quite inconvenient for the user to receive a message mentioning a wrong identifier for something that until now it was considered a string literal.

To conclude, if we think that the percentage of users/queries affected is rather low, and since ES-EQL is experimental, doesn't seem so impactful. Imho though, I have a preference over the more clear 2 step approach (1. error message for single quotes around string, 2. replacement of backquotes with single quote for identifiers).

**Comment by matriv** [comment]
Based on @astefan 's comment here: https://github.com/elastic/elasticsearch/issues/62650#issuecomment-695995357
I'd like to conclude that if we lock ourselves in the situation that we have to wait replacing the backquote with the single quote for identifiers until next major release, I'd rather proceed with this breaking change now, since it's safe enough (no wrong results for deprecated syntax).

**Comment by costin** [comment]
This PR can be closed without merging (see https://github.com/elastic/elasticsearch/issues/62663#issuecomment-699866306).
The issue of quoting an identifier containing the quoting character needs to be verified and addressed.

**Comment by matriv** [comment]
Closing this, see comments above and on the relevant issue: #62539

**Pull request closed without merging**